### PR TITLE
Fixing robot upgrades not being removed from lists all times.

### LINKED ISSUE #54038: Removing KA upgrades as miner borg does not remove them from the upgrades list (likely oversight)
## Reproduction:
Go mining module, get upgrades for your KA, then prybar them out and see that the list does not delete previously-installed upgrades. This is probably an oversight and isn't game-breaking. 
image

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
See #54038. Kinetic accelerators are suffered from a similar oversight.

## Why It's Good For The Game
This will fix edge cases and close #54038.

## Changelog
N/A

### CODE REVIEW

**Comment by Rohesie** [comment]
The removal of the getter is good, but I'd rather we handled this differently. Something like this:
```
/mob/living/silicon/robot/proc/add_to_upgrades(obj/item/borg/upgrade/new_upgrade)
  if(new_upgrade in upgrades)
    return
  upgrades += new_upgrade
  RegisterSignal(new_upgrade, COMSIG_MOVABLE_MOVED, .proc/on_upgrade_move)

/mob/living/silicon/robot/proc/remove_from_upgrades(obj/item/borg/upgrade/old_upgrade)
  if(!(old_upgrade in upgrades))
    return
  upgrades -= old_upgrade
  RegisterSignal(old_upgrade, COMSIG_MOVABLE_MOVED)

/mob/living/silicon/robot/proc/on_upgrade_move(obj/item/borg/upgrade/source, OldLoc, Dir, Forced)
  (...)
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -490,8 +486,35 @@
...
+	RegisterSignal(defib_instance, COMSIG_PARENT_QDELETING, .proc/on_defib_instance_qdel)
+
+/obj/item/borg/upgrade/defib/backpack/proc/on_defib_instance_qdel(obj/item/defibrillator/D)
+	defib_instance = null
+	qdel(src)
+
+/obj/item/borg/upgrade/defib/backpack/Destroy()
+	if(defib_instance)
+		QDEL_NULL(defib_instance)
+	. = ..()
```
Comment (Rohesie) [inline code/game/objects/items/robot/robot_upgrades.dm:510]:
```suggestion
	return ..()
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -490,8 +486,35 @@
...
+
+/obj/item/borg/upgrade/defib/backpack/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		if(defib_instance)
+			defib_instance.forceMove(R.drop_location())
```
Comment (Rohesie) [inline code/game/objects/items/robot/robot_upgrades.dm:516]: Won't this get deleted on destroy anyway?

Comment (Ghommie) [reply code/game/objects/items/robot/robot_upgrades.dm]: fixed.

**Pull request merged**